### PR TITLE
[4.0] Template Atum - Change Main Color Names and Variables

### PR DESCRIPTION
**Pull request by coolcat-creations**
### Summary of Changes
Renamed everything what has in common with the bg-dark color to Primary colour and the special color to secondary color. Moved the secondary color into the first column ob the backend settings.

### Testing Instructions
Apply the patch
npm run build:css
The color names in the setting should be now primary and secondary color and everything should work just like before the patch.

grafik

### Reason of the change

Every modern framework uses the terms _Primary Color_ and _Secondary color_ and for increased usability I suggest to change the naming in Joomla. 

Also it does not fix a color to have a specific brightness. Since the dark sidebar is gone and we don't need anymore to calculate all the shades of the sidebar for a11y another step would be also to remove the hue slider and replace it by a colorfield where the user is free to chose the color.

### CODE REVIEW

File: administrator/language/en-GB/tpl_atum.ini [hunk 1]
```diff
@@ -9,15 +9,15 @@ TPL_ATUM_ALTTEXT_SITE_LOGO_LABEL="Alt Text Site Logo"
 TPL_ATUM_ALTTEXT_SMALL_LOGO_LABEL="Alt Text Small Site Logo"
 TPL_ATUM_BACKEND_LOGIN="Joomla Administrator Login"
 TPL_ATUM_BACK_TO_CONTROL_PANEL="Back to Dashboard"
-TPL_ATUM_COLORS_HUE="Choose your hue value for the dark template colour"
-TPL_ATUM_COLORS_SETTINGS_BG_DARK_LABEL="Dark Background"
+TPL_ATUM_COLORS_HUE="Choose your hue value for the primary template colour"
```
Comment (brianteeman) [inline administrator/language/en-GB/tpl_atum.ini]:
```suggestion
TPL_ATUM_COLORS_HUE="Choose the hue value for the primary template colour"
```

I think that reads better

File: administrator/templates/atum/scss/_variables.scss [hunk 2]
```diff
@@ -46,7 +46,7 @@ $theme-colors: map-merge((
   dark:                            $gray-900,  //used in bootstrap and for table-border
   atum-link-color:                 $light-blue,
   atum-text-dark:                  $gray-700,
-  atum-bg-dark:                    $base-color, //standard background color
+  atum-primary-color:                    $base-color, //standard background color
```
Comment (brianteeman) [inline administrator/templates/atum/scss/_variables.scss]: Please remove the extra spaces

File: administrator/templates/atum/templateDetails.xml [hunk 3]
```diff
@@ -62,6 +62,15 @@
 					saveFormat="hsl"
 					default="hsl(214,63%,20%)"
 				/>
+
+				<field
+						name="secondary-color"
+						type="color"
+						label="TPL_ATUM_COLORS_SETTINGS_SECONDARY_COLOR_LABEL"
+						default="#001B4C"
+						filter="color"
```
Comment (brianteeman) [inline administrator/templates/atum/templateDetails.xml]:
```suggestion
					name="secondary-color"
					type="color"
					label="TPL_ATUM_COLORS_SETTINGS_SECONDARY_COLOR_LABEL"
					default="#001B4C"
					filter="color"
```

**Comment by brianteeman** [comment]
> The color names in the setting should be now primary and secondary color
Sensible change - thanks

**Comment by Quy** [comment]
Why not use `TPL_ATUM_COLORS_SETTINGS_PRIMARY_COLOUR_LABEL` instead of the long sentence for a label?

**Comment by coolcat-creations** [comment]
> Why not use `TPL_ATUM_COLORS_SETTINGS_PRIMARY_COLOUR_LABEL` instead of the long sentence for a label?

Actually as said above I would rather replace the hue field by really a primary colour field so this would be solved with another PR. The hue made sense when we calculated colors for the sidebar - now quite not anymore I think.

File: administrator/language/en-GB/tpl_atum.ini [hunk 4]
```diff
@@ -9,15 +9,15 @@ TPL_ATUM_ALTTEXT_SITE_LOGO_LABEL="Alt Text Site Logo"
...
+TPL_ATUM_COLORS_HUE="Choose the hue value for the primary template colour"
+TPL_ATUM_COLORS_SETTINGS_PRIMARY_COLOUR_LABEL="Primary Colour"
 TPL_ATUM_COLORS_SETTINGS_BG_LIGHT_LABEL="Light Background"
 TPL_ATUM_COLORS_SETTINGS_CONTRAST_COLOR_LABEL="Contrast Colour"
 TPL_ATUM_COLORS_SETTINGS_LABEL="Colour Settings"
 TPL_ATUM_COLORS_SETTINGS_LINK_COLOR_LABEL="Link Colour"
 TPL_ATUM_COLORS_SETTINGS_LINK_SIDEBAR_COLOR_LABEL="Sidebar Link Colour"
 TPL_ATUM_COLORS_SETTINGS_MONOCHROME_LABEL="Set Colour to Monochrome"
-TPL_ATUM_COLORS_SETTINGS_SPECIAL_COLOR_LABEL="Special Colour"
+TPL_ATUM_COLORS_SETTINGS_SECONDARY_COLOR_LABEL="Secondary Colour"
```
Comment (Quy) [inline administrator/language/en-GB/tpl_atum.ini:20]:
```suggestion
TPL_ATUM_COLORS_SETTINGS_PRIMARY_COLOUR_LABEL="Primary Colour"
TPL_ATUM_COLORS_SETTINGS_SECONDARY_COLOR_LABEL="Secondary Colour"
```

File: administrator/language/en-GB/tpl_atum.ini [hunk 5]
```diff
@@ -9,15 +9,16 @@ TPL_ATUM_ALTTEXT_SITE_LOGO_LABEL="Alt Text Site Logo"
 TPL_ATUM_ALTTEXT_SMALL_LOGO_LABEL="Alt Text Small Site Logo"
 TPL_ATUM_BACKEND_LOGIN="Joomla Administrator Login"
 TPL_ATUM_BACK_TO_CONTROL_PANEL="Back to Dashboard"
-TPL_ATUM_COLORS_HUE="Choose your hue value for the dark template colour"
-TPL_ATUM_COLORS_SETTINGS_BG_DARK_LABEL="Dark Background"
+TPL_ATUM_COLORS_HUE="Choose the hue value for the primary template colour"
+TPL_ATUM_COLORS_SETTINGS_PRIMARY_COLOUR_LABEL="Primary Colour"
```
Comment (Quy) [inline administrator/language/en-GB/tpl_atum.ini]:
```suggestion
```

**Comment by brianteeman** [comment]
@coolcat-creations please can you fix the conflicts

**Comment by uglyeoin** [comment]
@coolcat-creations we need conflicts fixed so we can test this.  Thanks Hurricane :)

**Comment by chmst** [comment]
@coolcat-creations as we have BS Version 5 - is this still valid? If yes: could you please resolve the conflicts?

**Comment by pritam825** [comment]
@coolcat-creations can you please solve the conflicts, so that we can test this pr. Thanks

**Comment by coolcat-creations** [comment]
Hey, can anyone direct me to how to resolve conflicts?  The button is disabled.
grafik

**Comment by Quy** [comment]
This may be considered an improvement and not a bug. It would be easier to close the PR and do it for the 4.2 branch.

**Comment by Quy** [comment]
This may be considered an improvement and not a bug. It would be easier to close the PR and do it for the 4.2 branch.

**Comment by coolcat-creations** [comment]
It's on my plan to redo this PR next week. I tried to merge 4.1 into this but it really seems too complicated...

**Comment by Quy** [comment]
Closing. To redo in v4.2.

**Pull request closed without merging**